### PR TITLE
added: tests on reading operations for plane dao and airport dao, rem…

### PR DESCRIPTION
**Pull request by klysov**
…oved extractFromResultingSet from interfaces, fixed NPE, implemented hashCode and equals

### CODE REVIEW

File: src/main/java/dao/PlaneDaoImpl.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public Plane getPlaneById(int id) {
         } catch (SQLException ex) {
             ex.printStackTrace();
         }
-        return null;
+        return new Plane();
```
Comment (SMishAlex) [inline src/main/java/dao/PlaneDaoImpl.java]: how can you return dao if you don't have entity in db? that's why I said that it's better don't catch exception here

File: src/main/java/entities/Airport.java [hunk 2]
```diff
@@ -5,23 +5,23 @@
  * @author klysov
  */
 public class Airport {
-    private int idAirport;
+    private int airportId;
     private String Name;
```
Comment (SMishAlex) [inline src/main/java/entities/Airport.java]: naming convention (your field's name can't start with capital letter)

File: src/main/java/entities/Airport.java [hunk 3]
```diff
@@ -5,23 +5,23 @@
...
 
     public Airport() {
     }
 
     public Airport(int idAirport, String name) {
-        this.idAirport = idAirport;
+        this.airportId = idAirport;
```
Comment (SMishAlex) [inline src/main/java/entities/Airport.java]: you should change param name too

File: src/main/java/entities/Airport.java [hunk 4]
```diff
@@ -5,23 +5,23 @@
...
         Name = name;
     }
 
     public int getIdAirport() {
```
Comment (SMishAlex) [inline src/main/java/entities/Airport.java]: and names of getters and setters too

File: src/main/java/entities/Plane.java [hunk 5]
```diff
@@ -49,4 +49,24 @@ public String toString() {
                 ", currentLoad=" + currentLoad +
```
Comment (SMishAlex) [inline src/main/java/entities/Plane.java]: planeId = idPlane... change naming here too

File: src/main/java/entities/Plane.java [hunk 6]
```diff
@@ -49,4 +49,24 @@ public String toString() {
...
+        return getCurrentLoad() == plane.getCurrentLoad();
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getIdPlane();
+        result = 31 * result + getMaxLoad();
+        result = 31 * result + getCurrentLoad();
+        return result;
+    }
```
Comment (SMishAlex) [inline src/main/java/entities/Plane.java]: do we really need hashCode method in our project? I mean will we ever use it?

File: src/test/java/dao/AirportDaoImplTest.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    @Test
+    public void testGetAirportById() {
+        AirportDaoImpl airportDao = new AirportDaoImpl();
+        assertEquals(new Airport(1, "Berlin"), airportDao.getAirportById(1));
+        assertEquals(new Airport(2, "London"), airportDao.getAirportById(2));
+        assertEquals(new Airport(3, "Paris"), airportDao.getAirportById(3));
+        assertEquals(new Airport(4, "Rome"), airportDao.getAirportById(4));
+    }
```
Comment (SMishAlex) [inline src/test/java/dao/AirportDaoImplTest.java]: you should test all CRUD methods, also your tests check if base fill correct by your sql script... normally it should contains some setup method where you put some data to DB and then test if db contains this data and dao work correctly with it.

**Review by SMishAlex — COMMENTED** [review]
and what about my advice about same method name for all dao? (not getAllAirports but just getAll)

**Comment by klysov** [comment]
added more: read commit description

File: src/main/java/MainServlet.java [hunk 8]
```diff
@@ -22,14 +22,14 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
 
         Connection connection = ConnectionFactory.getConnection();
-        AirportDaoImpl airport = new AirportDaoImpl();
-        PlaneDaoImpl plane = new PlaneDaoImpl();
-        DAO<Customer> customer = new CustomerDAOImpl();
+        AirportDAO airport = new AirportDAO();
+        PlaneDAO plane = new PlaneDAO();
```
Comment (SMishAlex) [inline src/main/java/MainServlet.java]: use interface where it's possible

File: src/main/java/entities/Airport.java [hunk 9]
```diff
@@ -1,42 +1,16 @@
...
-        return "Airport{" +
-                "airportId=" + idAirport +
-                ", Name='" + Name + '\'' +
-                '}';
-    }
+@Data
+@EqualsAndHashCode(exclude={"airportId"})
+public class Airport {
+    private int airportId;
+    private String name;
```
Comment (SMishAlex) [inline src/main/java/entities/Airport.java:15]: looks better, dosen't it?

File: src/main/java/services/ConnectionFactory.java [hunk 10]
```diff
@@ -10,7 +10,7 @@
  * @author klysov
  */
 public class ConnectionFactory {
-    private static final String URL = "jdbc:mysql://localhost:3306/lowcostairlines";
+    private static final String URL = "jdbc:mysql://localhost:3306/ticketsystem";
```
Comment (SMishAlex) [inline src/main/java/services/ConnectionFactory.java:13]: so what is the name of database?

File: src/test/java/dao/AirportDAOTest.java [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+
+public class AirportDAOTest {
+    private static ArrayList<String> testEntries = new ArrayList<>();
+    private static final int NUMBER_OF_ENTRIES = 10;
+
+    static {
+        for (int i = 0; i < NUMBER_OF_ENTRIES; i++) {
+            testEntries.add("ENTRY#" + i);
+        }
+    }
```
Comment (SMishAlex) [inline src/test/java/dao/AirportDAOTest.java]: I don't get what it is?

Comment (SMishAlex) [reply src/test/java/dao/AirportDAOTest.java]: any way it's better use `@BeforClass` instead of static initialization block

**Comment by klysov** [comment]
implemented previous advices

**Pull request merged**